R/utils.R

```diff
@@ -119,3 +119,33 @@ create_eml_contributors <- function(contributors) {
     onlineUrl = .x$path
   ))
 }
+
+#' Recursively remove NA values from a nested list
+#'
+#' Removes all elements that are NA at any depth. It works recursively, so any
+#' NA inside sublists will also be removed.
+#'
+#' @param nested_list A (possibly nested) list to clean of NA values.
+#' @return A list of the same structure as nested_list, but with all NA elements
+#' removed.
+#' @family helper functions
+#' @noRd
+#' @examples
+#' test_list <- list(
+#'   a = NA,
+#'   b = list(
+#'     c = 1,
+#'     d = NA
+#'   ),
+#'   e = 2
+#' )
+remove_na_recursive <- function(nested_list) {
```

Comment:
I would borrow the more generic helper function `clean_list` from frictionless, see https://github.com/frictionlessdata/frictionless-r/blob/b7fce08225ba2bd4bf5ae2f5be438f225fecfe84/R/utils.R#L33C1-L44